### PR TITLE
Update ClusterEnvironment documentation

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 ### The Solution
 
+- Segregated logical environments that can span across different kubernetes clusters
```
Comment (sadlerap) [inline README.md:20]: I'm not a fan of `Segregated` here; it's got some connotations, some of which aren't all that positive.
```suggestion
- Separated logical environments that can span across different kubernetes clusters
```

File: docs/entities/clusterenvironment.md [hunk 2]
```diff
@@ -78,13 +86,17 @@ status:
 
 ### Creation
 
-When a Cluster Environment is created, Primaza must verify the connection and update the field `status.state` accordingly.
+When a Cluster Environment is created, Primaza verifies the connection to the cluster and, if the connection test passes, its permissions in application and service namespaces.
```
Comment (dperaza4dustbit) [inline docs/entities/clusterenvironment.md:89]: This sentence is not clear did you mean something like this:
When a Cluster Environment is created, Primaza verifies the connection to the cluster, whether the connection test passes,  and its permissions to application and service namespaces.

File: docs/entities/clusterenvironment.md [hunk 3]
```diff
@@ -78,13 +86,17 @@ status:
...
 
 ### Deletion
 
-No operations should be performed.
+When a Cluster Environment is deleted, the permissions granted in Primaza's namespace to Users associated to namespace agents and agent deployments on target cluster's namespaces are removed.
 
 ### Update
 
-When a Cluster Environment is updated, Primaza must verify the connection and update the field `status.state` accordingly.
+As on [creation](#creation), Primaza verifies the connection and its permissions and reconciles application and service namespaces.
```
Comment (dperaza4dustbit) [inline docs/entities/clusterenvironment.md:100]: Something like this: As on [creation](#creation), Primaza verifies the connection and permissions to application and service namespaces.

**Review by dperaza4dustbit — CHANGES_REQUESTED** [review]

File: docs/entities/clusterenvironment.md (hunk 3 above)
Comment (filariow) [reply docs/entities/clusterenvironment.md:100]: What do you think about the following?

```suggestion
As on [creation](#creation), Primaza verifies the connection to and its permissions into the target cluster. Finally, it pushes agents in cluster's application and service namespaces.
```

File: docs/entities/clusterenvironment.md (hunk 2 above)
Comment (dperaza4dustbit) [reply docs/entities/clusterenvironment.md:89]: looks good to me

**Review by dperaza4dustbit — APPROVED** [review]
lgtm

**Review by dperaza4dustbit — APPROVED** [review]
lgtm

**Pull request merged**